fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/CustomModeledArmorRegistry.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import net.minecraft.item.Item;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.impl.client.item.CustomModeledArmorRegistryImpl;
+
+@Environment(EnvType.CLIENT)
+public interface CustomModeledArmorRegistry extends CustomModeledArmorProvider {
+	CustomModeledArmorRegistry INSTANCE = CustomModeledArmorRegistryImpl.INSTANCE;
```

Comment:
I personally want to move away from the `instance` design of this. Personally I think we should make this entirely static so we chop off the `INSTANCE` call.

Also why is this itself an armor provider. That should really be impl detail.

Reply:
> Also why is this itself an armor provider. That should really be impl detail.
why not, it's just duplicate methods, but if we are moving away from `INSTANCE` this can go